### PR TITLE
#160618118 Reporting an article

### PR DESCRIPTION
**Pull request by VictorUkafor**
### What does this PR do?
This pull request allows users to report articles that violate terms of agreement

### Description of Task to be completed?

- Write tests for the  reporting articles
- Write a method for posting a report on article
- Write a method for fetching all the reports of articles by the admin

### Any background context you want to provide?
N/A

### What are the relevant pivotal tracker stories?
#160618118

### Screenshots (if appropriate)
N/A

### Questions:
None

### CODE REVIEW

File: tests/reportArticleTest.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+        .send({ violation: 'plagiarisism' })
+        .send(article)
+        .end((err, res) => {
+          expect(res.status).to.equal(201);
+          expect(res.body.message).to.equal('You have reported this article successfully');
+          done();
+        });
+    });
+
+    it('Should report an article successfully when field is not field', (done) => {
```
Comment (Phunmbi) [inline tests/reportArticleTest.js:111]: Nice job with the tests, But this test and the one above look exactly the same but with different names.

File: tests/reportArticleTest.js [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+    });
+
+    it('Should return error when user is not authorized', (done) => {
+      chai
+        .request(app)
+        .get('/api/admin/reportedArticles')
+        .set({ authorization: userToken, Accept: 'application/json' })
+        .end((err, res) => {
+          expect(res.status).to.equal(403);
+          expect(res.body.message).to.equal('Access Denied!, You are not authorized to do this');
```
Comment (Phunmbi) [inline tests/reportArticleTest.js:144]: I think this error message can be a bit more descriptive.

File: tests/reportArticleTest.js (hunk 1 above)
Comment (VictorUkafor) [reply tests/reportArticleTest.js:111]: Thanks for pointing out this, I'll fix it now

File: tests/reportArticleTest.js (hunk 2 above)
Comment (VictorUkafor) [reply tests/reportArticleTest.js:144]: Noted

**Pull request merged**